### PR TITLE
Deploying Python Functions Into Web-Apps Using Gradio

### PR DESCRIPTION
**Pull request by ganning127**
**Name**: Ganning Xu

**Do you run a club?** Yes, at the North Carolina School of Science and Math (I just graduated high school though)

**What relevant experiences do you have to help you write this Jam?** The first language I learned (and my favorite language) is Python. I also did a natural language machine learning research project in Python (though this workshop requires no machine learning, and is all intuitive!)

**Will you be available on Slack to provide support to Hack Clubbers who run into issues while going through your Jam?** Of course, happy to help :)

### CODE REVIEW

File: jams/python-text-summarization-and-gradio.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Text Summarization in Python, Deploying the Model with Gradio
+
+### Details:
+_Outcome:_ Create a fully functional text summarization web-application from scratch using Python and Gradio. Anyone can use the model you create using
```
Comment (sarthaktexas) [inline jams/python-text-summarization-and-gradio.md]: using...?

File: jams/python-text-summarization-and-gradio.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Text Summarization in Python, Deploying the Model with Gradio
```
Comment (sarthaktexas) [inline jams/python-text-summarization-and-gradio.md]: Can you write a quick explanation on what Gradio is? I know you can Google it but not everyone will when they see this jam.

File: jams/python-text-summarization-and-gradio.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+### Details:
+_Outcome:_ Create a fully functional text summarization web-application from scratch using Python and Gradio. Anyone can use the model you create using 
+
+_Requirements:_ Google Colab Access (comes free with Google/Gmail account)
```
Comment (sarthaktexas) [inline jams/python-text-summarization-and-gradio.md:8]: This is a great tool, I'm excited to see a proposal using this :)

File: jams/python-text-summarization-and-gradio.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+_Time estimation:_ 55 mins (created a video workshop about this same topic, and it was 53 minutes)
+
+### Breakdown
+- Part 0: slides that cover the basics of natural language processing (how computers understand human language) and different approaches to text summarization (in an intuitive way)
+- Part 1: Get everyone onto Google Colab
+- Part 2: Code out the Python text summarization function (input is text, returns summary)
+- Part 3: Deploy this using Gradio (is only a couple lines of code, hooray!)
+- Part 4: Customize the model/Gradio web app to your liking
+- Hooray!!! Now you've got a text summarization web app!
```
Comment (sarthaktexas) [inline jams/python-text-summarization-and-gradio.md]: I really, really love this as a workshop, but this doesn't provide for customization. Look at above comment.

File: jams/python-text-summarization-and-gradio.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
+# Text Summarization in Python, Deploying the Model with Gradio
+
+### Details:
+_Outcome:_ Create a fully functional text summarization web-application from scratch using Python and Gradio. Anyone can use the model you create using 
+
+_Requirements:_ Google Colab Access (comes free with Google/Gmail account)
+
+_Language used:_ Python
+
+_Customization opportunity:_ We'll cover one method of using Python for text summarization (which doesn't technically involve machine learning), but there are many other methods that can be used (including machine learning). Additionally, you can customize the Gradio web app to show different metrics about the text you want summarized (like word count, character count, sentence count, etc), in addition to the summary.
```
Comment (sarthaktexas) [inline jams/python-text-summarization-and-gradio.md]: This is a great start. It seems like a better use for this jam would be using Gradio as a dashboard for machine learning opportunities. Maybe cover that as a general Python concept, and then use gradio to display data from the model? Not sure if this makes sense, you can ping me on slack if need be.

**Review by sarthaktexas — CHANGES_REQUESTED** [review]
I love this idea, just needs more room for customization. Outline is presented beautifully and I would be very excited to run this in my club as a club leader. Thanks for the proposal!

**Comment by ganning127** [comment]
Just added your requested changes from our meeting, would love some more feedback

**Review by sarthaktexas — APPROVED** [review]
100%, I love this jam Ganning

File: jams/python-text-summarization-and-gradio.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+_Time estimation:_ 50 minutes
+
+### Breakdown
+
+- Part 0: Get everyone onto Google Colab (5 mins)
+- Part 1: basics/cool functionality of Python (if-statements, functions, API calls, etc) (15 mins)
+- Part 2: Give everyone 15 minutes to code out a Python function of their chioce (15 mins)
+- Part 3: Deploy this using Gradio, and explain various gradio input types (text, image, file, radio-select, etc) (15 mins)
+- Part 4 (extension): If you want a permanent deployed link (longer than the default 72 hours), deploy your Gradio model to HuggingFace in the terminal.
```
Comment (Arpan-206) [inline jams/python-text-summarization-and-gradio.md]: How much time is this extension expected to take?

Comment (Arpan-206) [reply jams/python-text-summarization-and-gradio.md]: Would be wonderful to factor that in as well.

**Review by Arpan-206 — CHANGES_REQUESTED** [review]
Just a minor query, looks good otherwise!

File: jams/python-text-summarization-and-gradio.md (hunk 6 above)
Comment (ganning127) [reply jams/python-text-summarization-and-gradio.md]: Great point! It should take around 10 minutes. They would need to make a huggingface account, create a new space, and just copy and paste their code from Colab into HuggingFace. Then, it's automatically running :) 

I've added a commit that adds this to the .md file

**Review by Arpan-206 — APPROVED** [review]
Wonderful Jam, looking forward to using this in my club!

**Review by faisalsayed10 — APPROVED** [review]
LOVE THIS! quite a unique jam!

**Pull request merged**